lib/OpenQA/WebAPI/Controller/API/V1/Table.pm

```diff
@@ -397,9 +397,14 @@ sub _prepare_settings {
     my @keys;
     if ($hp->{settings}) {
         for my $k (keys %{$hp->{settings}}) {
-            $k = trim $k;
             my $value = trim $hp->{settings}->{$k};
-            $k =~ s/[^\]\[0-9a-zA-Z_\+]//g;
+            $k = trim $k;
+            my %invalid;
+            () = $k =~ s/([^\]\[0-9a-zA-Z_\+])/$invalid{$1} = 1; ''/egr;
```

Comment:
Ok, I didn't see that you would not need the returned string.
Clever :) but maybe then just use a match:
```suggestion
            @inv{$foo =~ m/([^\]\[0-9a-zA-Z_\+])/g} = ();
```